tensorboard/plugins/histogram/summary_v2.py

```diff
@@ -196,53 +196,50 @@ def _buckets(data, bucket_count=None):
         tf.debugging.assert_type(bucket_count, tf.int32)
         data = tf.reshape(data, shape=[-1])  # flatten
         data = tf.cast(data, tf.float64)
-        is_empty = tf.equal(tf.size(input=data), 0)
-
-        def when_empty():
+        data_size = tf.size(input=data)
+        if tf.equal(data_size, 0):
             return tf.constant([], shape=(0, 3), dtype=tf.float64)
 
-        def when_nonempty():
-            min_ = tf.reduce_min(input_tensor=data)
-            max_ = tf.reduce_max(input_tensor=data)
-            range_ = max_ - min_
-            is_singular = tf.equal(range_, 0)
-
-            def when_nonsingular():
-                bucket_width = range_ / tf.cast(bucket_count, tf.float64)
-                offsets = data - min_
-                bucket_indices = tf.cast(
-                    tf.floor(offsets / bucket_width), dtype=tf.int32
-                )
-                clamped_indices = tf.minimum(bucket_indices, bucket_count - 1)
-                one_hots = tf.one_hot(clamped_indices, depth=bucket_count)
-                bucket_counts = tf.cast(
-                    tf.reduce_sum(input_tensor=one_hots, axis=0),
-                    dtype=tf.float64,
-                )
-                edges = tf.linspace(min_, max_, bucket_count + 1)
-                # Ensure edges[-1] == max_, which TF's linspace implementation does not
-                # do, leaving it subject to the whim of floating point rounding error.
-                edges = tf.concat([edges[:-1], [max_]], 0)
-                left_edges = edges[:-1]
-                right_edges = edges[1:]
-                return tf.transpose(
-                    a=tf.stack([left_edges, right_edges, bucket_counts])
-                )
-
-            def when_singular():
-                center = min_
-                bucket_starts = tf.stack([center - 0.5])
-                bucket_ends = tf.stack([center + 0.5])
-                bucket_counts = tf.stack(
-                    [tf.cast(tf.size(input=data), tf.float64)]
-                )
-                return tf.transpose(
-                    a=tf.stack([bucket_starts, bucket_ends, bucket_counts])
-                )
+        min_ = tf.reduce_min(input_tensor=data)
+        max_ = tf.reduce_max(input_tensor=data)
+        range_ = max_ - min_
+        is_singular = tf.equal(range_, 0)
+
+        bucket_width = range_ / tf.cast(bucket_count, tf.float64)
+        offsets = data - min_
+        # For singular data, indices should simply be
```

Comment:
This is certainly true, but if we're going to use a tf.cond() anyway, we can probably push this a bit further downstream to avoid the overhead of generating all the indices only to reduce_sum() everything later.  I.e. we can just directly construct something like `single_value_bucket_counts` as `bucket_count - 1` zeros followed by a single count equal to `data_size`, and use that instead of doing the computation of the bucket counts.